Civi/WorkflowMessage/GenericWorkflowMessage.php

```diff
@@ -117,4 +118,17 @@ protected function exportExtraTokenContext(array &$export): void {
     $export['smartyTokenAlias']['taxTerm'] = 'domain.tax_term';
   }
 
+  /**
+   * Additional text to include in the receipt.
+   *
+   * @var string
+   *
+   * @scope tplParams as user_text
+   */
+  protected $userEnteredText;
+
+  public function getUserEnteredText(): ?string {
+    return $this->userEnteredText ? htmlentities($this->userEnteredText): NULL;
```

Comment:
not sure if this is best - but we should probably do some form of escaping

Reply:
This would apply to text templates too, right? So I don't think we can (unless there's something that decodes them for that somewhere).

Reply:
I don't think I understand this as a generic thing that applies to *all* message types. It smells more like a reusable `trait`.

Re: escaping - the main thing is to be clear about the intended format: plain text  or rich text (HTML)? 

* If content comes from an existing source, then it's an empirical question - what is that source? (If it's new content, then I guess it's discretionary - but still needs to be explicit.)
* If the content is plain-text, then any HTML email generator needs HTML-ization -- escaping at a minimum and/or maybe some newline handling. (Plain text email generators won't that stuff.)
* If the content is rich-text, then any HTML email generator should use a purifier. (Plain text email generators should use `striptags` or `html2text`.)

Reply:
Hmm - I was thinking this was something generic enough to put in the 'all template' category - so that any workflow template from a form could include user content. Currently not all of them do - but where they do then I think any text could be entered

Reply:
I think moving towards being able to add user content to pretty much any message template being sent would be a good thing.

Reply:
On the HTML versus plain text issue, see also #27173. I think it would make sense to have the user_text and the various receipt_texts all be HTML in the long run, then strip for plain text emails. It should be possible to use links in both the user_text and the receipt_texts. Our users have definitely complained about this in the past.